requirements/skinny-requirements.yaml

```diff
@@ -79,3 +79,8 @@ pydantic:
   pip_release: pydantic
   minimum: "1.0"
   max_major_version: 2
+
+typing-extensions:
+  pip_release: typing-extensions
+  minimum: "4.0.0"
+  max_major_version: 4
```

Comment:
As mlflow already depends on `pydantic` which depends on `typing-extensions`, this doesn't add any extra dependency.

Reply:
If so, why do we need to add this?

Reply:
because we should not depend on transient dependencies in general.

Reply:
What if pydantic suddenly says "Hey, we don't need typing_extensions anymore."?

Reply:
I also don't want us and external contributors to break mlflow by `typing_extensions` imports.

Reply:
Then we need to be careful about compatibility issue, typing_extensions version we pin v.s pydantic pins

Reply:
pydantic has a more strict requirement, but yeah we should be careful.